validator/rules/rules.go

```diff
@@ -54,29 +57,42 @@ func NewDefaultRules() *Rules {
 	return r
 }
 
+// AddRule adds a rule with the specified name and rule function to the rule set.
+// If a rule with the same name already exists, it will not be added.
 func (r *Rules) AddRule(name string, ruleFunc core.RuleFunc) {
 	if r.rules == nil {
 		r.rules = make(map[string]core.RuleFunc)
 	}
-	r.rules[name] = ruleFunc
+
+	if _, exists := r.rules[name]; !exists {
+		r.rules[name] = ruleFunc
+	}
 }
 
+// GetInner returns the internal rule map.
+// If the map is not initialized, it returns an empty map.
 func (r *Rules) GetInner() map[string]core.RuleFunc {
 	if r.rules == nil {
 		return make(map[string]core.RuleFunc)
 	}
 	return r.rules
 }
 
+// RemoveRule removes a rule with the specified name from the rule set.
+// If no rule with the specified name exists, it does nothing.
 func (r *Rules) RemoveRule(name string) {
 	if r.rules != nil {
 		delete(r.rules, name)
 	}
 }
 
+// ReplaceRule replaces a rule with the specified name with a new rule function.
+// If no rule with the specified name exists, it does nothing.
 func (r *Rules) ReplaceRule(name string, ruleFunc core.RuleFunc) {
 	if r.rules == nil {
 		r.rules = make(map[string]core.RuleFunc)
 	}
-	r.rules[name] = ruleFunc
+	if _, exists := r.rules[name]; exists {
```

Comment:
```suggestion
	if _, exists := r.rules[name]; !exists {
```
Based on your comment, I think you missed a character